src/sentry/web/frontend/idp_email_verification.py

```diff
@@ -0,0 +1,19 @@
+from django.http.response import HttpResponseNotFound, HttpResponseRedirect
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.auth.idpmigration import get_redis_key, verify_new_identity
+from sentry.utils.auth import get_login_url
+from sentry.web.frontend.base import BaseView
+
+
+class IDPEmailVerificationView(BaseView):
+    def confirm_email(request: Request, key: str) -> Response:
+        verification_key = get_redis_key(key)
+        if verify_new_identity(key):
+            request.session["verification_key"] = verification_key
```

Comment:
assuming you're making changes to this based on convo earlier today

Reply:
may be better to give it more descriptive name in the session: `confirm_account_verification_key`